src/planner/kinodynamic/KinodynamicTimer.cpp

```diff
@@ -0,0 +1,273 @@
+#include "aikido/planner/kinodynamic/KinodynamicTimer.hpp"
+#include <dart/dart.hpp>
+#include <aikido/common/Spline.hpp>
+#include <aikido/common/StepSequence.hpp>
+
+#include "Path.h"
+#include "Trajectory.h"
+
+namespace aikido {
+namespace planner {
+namespace kinodynamic {
+
+//==============================================================================
+std::unique_ptr<aikido::trajectory::Spline>
+createSplineFromWaypointsAndConstraints(
```

Comment:
My understanding of this function is that we're
- Calling this library to generate a smooth and retimed trajectory from a sequence of waypoints (from our planned trajectory)
- Sampling points at each timestep from their trajectory, then fitting a spline to this sequence of points

I haven't read the paper so I don't know what their trajectory representation is, but it's not clear why discarding that representation and refitting a spline is necessary. I'm also concerned that it doesn't seem that we're checking that our planning constraints remain satisfied after postprocessing.

Reply:
This is a great catch! "Theoretically", it is a timing method that won't change the geometry of a path. So the collision violation is not taken care of (so as for the parabolic timer), by assuming that it is done in the planning. However, it does introduce slight geometry changes in a preprocessing step, which uses circular blends to make a path differentiable. How much the change is controlled by a parameter `maxDeviation`. It is possible that the timing will fail because an unideal path geometry is provided. 

If a user choose a reasonable value for `maxDeviation`, it is very unlike that the planning constraints will be violated.

Reply:
Given that the max deviation can be upto half the distance of line segments between consecutive points, we might want to check for collisions even during this procedure. I am not sure if the assumption that planning constraints will not be violated is valid while `0 < maxDeviations < 0.5 x (edge Lengths)`.

Reply:
@aditya-vk Hmm, I agree. I suggest that we can have a post validation, which is a constraint validation for a spline trajectory. There is already a ticket about this #212 . I can work on that one. This will be useful to a few other functions.